utility/firebase.js

```diff
@@ -199,43 +199,33 @@ const fireDb = {
       SignUpText: signupText || ''
     });
   },
-  addSponsorInformation: async (website, sponsor) => {
+  setSponsor: async (website, sponsor) => {
     const ref = db
-      .collection(webCollection)
+      .collection("Hackathons")
       .doc(website)
-      .collection('Sponsors');
-    await ref.add({
-      image: sponsor.image,
+      .collection('Sponsors')
+      .doc(sponsor.name);
+    await ref.set({
+      imgURL: sponsor.image,
       name: sponsor.name,
-      url: sponsor.url,
-      rank: sponsor.rank,
-      altImage: sponsor.altImage
-    });
+      link: sponsor.link,
+      tier: sponsor.tier,
+      lastmod: sponsor.lastmod
+    })
   },
-  async deleteSponsor(website, image) {
-    let altImage = false;
-    try {
-      const sponsors = await db
-        .collection(webCollection)
-        .doc(website)
-        .collection('Sponsors')
-        .get();
-      sponsors.forEach(async sponsor => {
-        if (sponsor.data().image === image) {
-          altImage = !!sponsor.data().altImage;
-          await sponsor.ref.delete();
-        }
-      });
-    } catch (e) {
-      return false;
-    }
-    const ref = storage.ref(`${website}/${image}`);
+  deleteSponsor: async(website, sponsorName) => {
+    const ref = db
+      .collection("Hackathons")
+      .doc(website)
+      .collection("Sponsors")
+      .doc(sponsorName);
     await ref.delete();
-    if (altImage) {
-      const altRef = storage.ref(`${website}/alt${image}`);
-      await altRef.delete();
-    }
-    return true;
+  },
+  alexgetSponsors: async(website) => { 
+    var A = {};
+    const B = await db.collection("Hackathons").doc(website).collection("Sponsors").get();
+    B.docs.forEach(doc => {A[doc.id] = doc.data()})
+    return A;
```

Comment:
A more functional way to do this would be like this:
```javascript
const refs = await db.collection("Hackathons").doc(website).collection("Sponsors").get();
const docs = refs.map(doc => doc.data())
return docs;
```

also you can just delete the old get function and rename this one to getSponsors